pkg/util/k8sutil/k8sutil.go

```diff
@@ -266,6 +266,11 @@ func NewEtcdPod(m *etcdutil.Member, initialCluster []string, clusterName, state,
 			Volumes: []v1.Volume{
 				{Name: "etcd-data", VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}},
 			},
+			// DNS A record: [m.Name].[clusterName].Namespace.pod.cluster.local.
+			// For example, etcd-0000 in default namesapce will have DNS name
+			// `etcd-0000.etcd.default.pod.cluster.local`.
```

Comment:
Also I'd like to note that the DNS search for the pods in default NS includes `default.svc.cluster.local`, so you should still be able to lookup address of the member pod via `etcd-0000.etcd` from within the namespace. I don't think there is any reason to leave the service around.